docs/GDPs/gdp-0002.rst

```diff
@@ -0,0 +1,341 @@
+
+
+Backward Compatibility
+----------------------
+
+This GDP aims to be fully backward-compatible.
+
+
+Detailed Description
+--------------------
+
+The intent of this GDP is to a specify horizontal bounds inside the stencil in a similar fashion to how vertical intervals are specified.
+However, the fact that there are multiple dimensions in the horizontal, halos, and domain decomposition imply the feature needs to be more complex than a simple index range used for the vertical.
+
+As introduced above, we propose adding a new gtscript `region()` specifier that applies the stencil on a region of the space when a non-zero intersection with the global region specified occurs.
+The idea behind using the region objects is that they can incorporate both the iteration space a process sees and the global partitioning that an application must handle, all while hiding the complexity at the stencil level for model developers.
+In this proposal, the application remains in control of the partitioning data so **the partitioner is entirely transparent to both GT4Py and its backend.**
+
+To explain how this works, consider part of the example above with a domain of size N by N by K points, decomposed by three processes using vertical strips of the domain:
+
+.. math:: \mathrm{proc}\, 0: [0,n) \times [0,N) \times [0,K),\quad \mathrm{proc}\, 1: [n,2n) \times [0,N) \times [0,K),\quad \mathrm{proc}\, 2: [2n,N) \times [0,N) \times [0,K).
+
+.. code-block:: python
+
+    with region(WestEdge(0, 1), EastEdge(0, 1)), interval(...):
+
+This part of the example neatly indicates that it is intended to be run on both the east and west edges of horizontal plane for each `k`, with an index strip `1` wide.
+The construction and evaluation of these application-specific objects, `WestEdge(0,1)` and `EastEdge(0,1)`, are done before code generation, and can call into user-provided code.
+These objects have attributes `local_origin` and `extent` which define the iteration space in the horizontal plane `i` and `j`.
+The union of the boxes defined by each of these objects is used for iteration.
+
+When GT4Py asks for the iteration space from the region objects, it sees different `local_origin` and `extent` on each process using the local iteration space indices:
+
+.. code-block:: python
+
+    proc 0:    local_origin=(0, 0), extent=(1, N)
+    proc 1:    local_origin=(0, 0), extent=(0, 0)
+    proc 2:    local_origin=(n-1, 0), extent=(1, N)
+
+The result of the evaluation is inserted directly into code, so GT4Py and its backend never deal with partitioning.
+Translation to the local indices is done within the Box object specialization from the application.
+
+Note that if this example had only one process, the local and global iteration spaces coincide, so the `local_origin` and `extent` set simply get passed on to GT4Py unmodified.
```

Comment:
so if you only have one processor you get a set of two origin extent pairs?

Reply:
Yes. I added this example to the doc.